### PR TITLE
ci: upgrade the kubernetes versions for ci

### PR DESCRIPTION
**Pull request by subhamkrai**
since, kubernetes v1.28.0 release upgrading the
latest k8s version to v.1.28.0 and minimum version to v1.23.17

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
@thotz I see we are pulling you image in code `Pulling image "quay.io/jthottan/pythonwebserver:latest"` could you fix this?

File: .github/workflows/integration-test-helm-suite.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.22.17", "v1.27.2"]
+        kubernetes-versions: ["v1.23.17", "v1.28.0"]
```
Comment (travisn) [inline .github/workflows/integration-test-helm-suite.yaml:28]: Let's revert all the changes that increase the min version, for two purposes:
1. We can backport it to 1.12 and avoid some merge conflicts
2. When it is closer to the 1.13 release then we can increase the min version for that release.

**Review by travisn — CHANGES_REQUESTED** [review]
Great to see the CI is passing against K8s 1.28!

File: .github/workflows/integration-test-helm-suite.yaml (hunk 1 above)
Comment (subhamkrai) [reply .github/workflows/integration-test-helm-suite.yaml]: agreed, reverted

**Comment by subhamkrai** [comment]
@travisn I see ` FAIL: TestCephObjectSuite/TestWithoutTLS/run_a_second_object_store/create_CephObjectStore (203.18s)` test failed twice. So, I guess we need to fix this CI also with k8s v1.28.0  😓

**Comment by travisn** [comment]
> @travisn I see ` FAIL: TestCephObjectSuite/TestWithoutTLS/run_a_second_object_store/create_CephObjectStore (203.18s)` test failed twice. So, I guess we need to fix this CI also with k8s v1.28.0 😓

Let's see if it passes after the latest push. I wouldn't think it's related to 1.28. But if we do find it's consistently failing, for now we can revert moving to 1.28 just fro that test suite.

**Comment by thotz** [comment]
> @thotz I see we are pulling you image in code `Pulling image "quay.io/jthottan/pythonwebserver:latest"` could you fix this?
> 
> edit: also object suite is failing https://github.com/rook/rook/actions/runs/5891231184/job/15977871963?pr=12748
> 
> because
> 
> ```
>   Normal  Scheduled  3m46s  default-scheduler  Successfully assigned default/sample-http-server-67bc776d8f-fxsl4 to fv-az213-532
>   Normal  Pulling    3m42s  kubelet            Pulling image "quay.io/jthottan/pythonwebserver:latest"
>     ceph_bucket_notification_test.go:68: 
>         	Error Trace:	/home/runner/work/rook/rook/tests/integration/ceph_bucket_notification_test.go:68
>         	Error:      	Expected nil, but got: &errors.errorString{s:"Giving up waiting for pod with label app=sample-http-server in namespace default to be running"}
>         	Test:       	TestCephObjectSuite/TestWithTLS/create_HTTP_Endpoint_for_receiving_notifications
> 2023-08-17 13:36:36.604846 I | integrationTest: Skipping COSI driver test as TLS is enabled
> 2023-08-17 13:36:36.640228 I | integrationTest: successfully deleted store TLS secret
> 2023-08-17 13:36:36.662003 D | testutil: created log file: /home/runner/work/rook/rook/tests/integration/_output/tests/ObjectSuite_TestWithTLS_localhost_object-ns-system_rook-ceph-operator-6576499c89-gb8rb_1692279396.log
> 2023-08-17 13:36:37.942265 D | testutil: created log file: /home/runner/work/rook/rook/tests/integration/_output/tests/ObjectSuite_TestWithTLS_localhost_object-ns-system_rook-ceph-operator-6576499c89-gb8rb_previous_1692279397.log
> 2023-08-17 13:36:37.947988 D | exec: Running command: kubectl -n object-ns-system logs rook-ceph-operator-6576499c89-gb8rb -c rook-ceph-operator
> 2023-08-17 13:36:39.626766 I | testutil: will alert if any pods were restarted in namespace object-ns
> 2023-08-17 13:36:39.703984 I | testutil: number of time pod rook-ceph-mgr-a-57dc4f4f84-2pkh8 has restarted is 1
> 2023-08-17 13:36:39.704032 I | testutil: number of time pod rook-ceph-mon-a-74cbf9b657-sc8gl has restarted is 2
> 2023-08-17 13:36:39.704039 I | testutil: number of time pod rook-ceph-mon-c-dccc44bc7-rrbmr has restarted is 1
> === NAME  TestCephObjectSuite
>     k8s_helper.go:473: 
>         	Error Trace:	/home/runner/work/rook/rook/tests/framework/utils/k8s_helper.go:473
>         	            				/home/runner/work/rook/rook/tests/framework/installer/ceph_installer.go:643
>         	            				/home/runner/work/rook/rook/tests/framework/installer/ceph_installer.go:633
>         	            				/home/runner/work/rook/rook/tests/integration/ceph_object_test.go:90
>         	            				/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.4/suite/suite.go:205
>         	            				/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.4/suite/suite.go:216
>         	            				/home/runner/work/rook/rook/tests/integration/ceph_object_test.go:52
>         	Error:      	Not equal: 
>         	            	expected: 0
>         	            	actual  : 1
> ```
> 
> PTAL

Is the pull failing for that image??

**Comment by subhamkrai** [comment]
@thotz I have not checked that but 
```"Giving up waiting for pod with label app=sample-http-server in namespace default to be running``` the pod is not going to running sate

**Comment by travisn** [comment]
@subhamkrai How about rebasing again to get a fresh test run?

**Comment by subhamkrai** [comment]
> @subhamkrai How about rebasing again to get a fresh test run?

done

**Comment by travisn** [comment]
The object store is still failing on 1.28. How about reverting the object test to use 1.27 still so we can get this merged? We can track that with a separate issue.

**Review by travisn — APPROVED** [review]

**Pull request merged**